lib/index.d.ts

```diff
@@ -110,12 +118,26 @@ export class Index {
 
 }
 
+/**
+ * IndexFlat Abstract Index.
+ */
+export abstract class IndexFlat extends Index {
```

Comment:
This pattern is the cleanest way to emulate faiss abstractions w/o losing IndexBase template class (which would greatly complicate this library).